### PR TITLE
LanceDB Integration

### PR DESCRIPTION
**Pull request by pradh**
This PR adds support for LanceDB, a new on-disk embeddings DB, so we can support multiple, much larger embeddings without having them all loaded into memory.  LanceDB also supports quicker load times.

Changes:

* Implementation of the LanceDBStore.  A new index `lance_medium_ft` using that store is added for testing.

* Support to create LanceDB index in `build_embeddings` tool.

* A simple integration test for lancedb index that tests parity with memory index.

Size impact:  Adds about ~90MB of packages and one more dependency on torch

### CODE REVIEW

File: static/js/apps/explore/app.tsx [hunk 1]
```diff
@@ -269,6 +269,7 @@ export function App(props: { isDemo: boolean }): JSX.Element {
     const topic = getSingleParam(hashParams[URL_HASH_PARAMS.TOPIC]);
     const place = getSingleParam(hashParams[URL_HASH_PARAMS.PLACE]);
     const dc = getSingleParam(hashParams[URL_HASH_PARAMS.DC]);
+    const idx = getSingleParam(hashParams[URL_HASH_PARAMS.IDX]);
```
Comment (shifucun) [inline static/js/apps/explore/app.tsx:272]: what is this idx?

**Review by shifucun — APPROVED** [review]
very nice!

File: static/js/apps/explore/app.tsx (hunk 1 above)
Comment (pradh) [reply static/js/apps/explore/app.tsx:272]: This is the embeddings index listed in embeddings.yaml.  I added a comment.

Comment (pradh) [reply static/js/apps/explore/app.tsx:272]: Mainly for testing.

**Comment by pradh** [comment]
Thanks for the review!

**Pull request merged**